src/pages/plans/Plans.tsx

```diff
@@ -1,96 +1,241 @@
+  const [editPlan] = useUpdatePlanPriceMutation();
+  const [deletePlan] = useDeletePlanMutation();
+  const [createPlan] = useCreatePlanMutation();
   const { userId } = useAuthentication();
-  const { data: prices } = useGetUserPricesQuery(userId!);
-  const [newAmount, setNewAmount] = useState<number>(0);
-  const [selectedPriceId, setSelectedPriceId] = useState("");
-  const [updatePrice, { isLoading: isUpdating }] = useUpdatePriceMutation();
-  const selectedPrice = prices?.find((p) => p.id === selectedPriceId);
-  const handleAmountChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const rawValue = e.target.value.replace(/[^\d]/g, "");
-    setNewAmount(Number(rawValue) / 100);
-  };
-
-  const handleSubmit = (e: React.FormEvent) => {
-    e.preventDefault();
-    if (!selectedPriceId) {
-      toast.error("Por favor, selecciona un plan.");
-      return;
+  const {
+    data: plans,
+    isLoading,
+    isError,
+  } = useGetUserPlansQuery(userId ? userId : skipToken);
+  useMemo(() => {
+    if (plans) {
+      setPlanList(plans);
     }
+  }, [plans]);
+
+  //Filtro de busqueda por nombre desde el frontend, cuando
+  //este el backend se tiene que modificar:
+  const filteredPlans = useMemo(() => {
+    if (!filter) return planList;
 
-    if (!newAmount || isNaN(Number(newAmount))) {
-      toast.error("Ingresa un monto válido.");
-      return;
+    return planList.filter((plan) =>
+      plan.name.toLowerCase().includes(filter.toLowerCase())
+    );
+  }, [planList, filter]);
+
+  const handleClearFilters = () => {
+    setFilter("");
+  };
+  const handleMutation = async (
+    action: () => Promise<any>,
+    onSuccess: () => void,
+    successMessage: string,
+    errorMessage: string
+  ) => {
+    try {
+      await action();
+      toast.success(successMessage);
+      onSuccess();
+    } catch (error: any) {
+      toast.error(errorMessage);
     }
+  };
+  //CREAR PLAN
+  const handleCreateModal = async () => {
+    if (!formRef.current) return;
+
+    const data = await formRef.current.submit();
+    if (!data) return;
+
+    await handleMutation(
+      () => createPlan(data).unwrap(),
+      () => setShowCreateModal(false),
+      "Plan creado correctamente",
+      "Error al crear el plan"
+    );
+  };
+  //EDITAR PLAN
+  const handleEditModal = async () => {
+    if (!formRef.current) return;
+
+    const data = await formRef.current.submit();
+    if (!data) return;
 
-    updatePrice({
-      priceId: selectedPriceId,
-      amount: Number(newAmount),
-    })
-      .unwrap()
-      .then(() => {
-        toast.success("Precio actualizado con éxito");
-        setNewAmount(0);
-      });
+    await handleMutation(
+      () => editPlan(data).unwrap(),
+      () => setShowEditModal(false),
+      "Plan editado correctamente",
+      "Error al editar el plan"
+    );
   };
+  //ELIMINAR PLAN
+  const handleDeleteModal = async () => {
```

Comment:
Y aca, pareciera que estas creando, editando o eliminando un modal con los nombres, pero en realidad son acciones para los planes


```suggestion
  const handleDeletePlan = async () => {
```